### PR TITLE
Fixes deprecation warning in Compression.swift

### PR DESCRIPTION
**Pull request by valeriyvan**

### CODE REVIEW

**Comment by CodaFi** [comment]
rdar://63724629

@swift-ci smoke test

File: stdlib/public/Darwin/Compression/Compression.swift [hunk 1]
```diff
@@ -338,11 +338,11 @@ public class InputFilter<D: DataProtocol> {
     // Allocate result
     var result = Data(count: count)
 
-    try result.withUnsafeMutableBytes { (dst_ptr: UnsafeMutablePointer<UInt8>) in
```
Comment (lorentey) [inline stdlib/public/Darwin/Compression/Compression.swift:341]: The type annotation was only needed to force the type checker to choose the deprecated method here; please replace with a variable called `buffer`.

File: stdlib/public/Darwin/Compression/Compression.swift [hunk 2]
```diff
@@ -338,11 +338,11 @@ public class InputFilter<D: DataProtocol> {
...
+    try result.withUnsafeMutableBytes { (dst_ptr: UnsafeMutableRawBufferPointer) in
 
       // Write to result until full, or end reached
       _stream.dst_size = count
-      _stream.dst_ptr = dst_ptr
```
Comment (lorentey) [inline stdlib/public/Darwin/Compression/Compression.swift:345]: `bindMemory` is not the right call here. @atrick is this one a good usecase for `assumingMemoryBound`?

Comment (atrick) [reply stdlib/public/Darwin/Compression/Compression.swift:345]: That's right. This is a good use of `assumingMemoryBound(to:)`. The C struct imported as `UnsafeMutablePointer<UInt8>` points into Foundation.Data's bytes, which don't have an element type. It's up to the user here to make sure that anything pointing into that Data object's bytes has `UInt8` type.

`bindMemory` is more appropriate when Data's bytes were used with some other type, and now you need to start using it with pointers of a different type.

File: stdlib/public/Darwin/Compression/Compression.swift (hunk 1 above)
Comment (valeriyvan) [reply stdlib/public/Darwin/Compression/Compression.swift:341]: Fixed

File: stdlib/public/Darwin/Compression/Compression.swift (hunk 2 above)
Comment (valeriyvan) [reply stdlib/public/Darwin/Compression/Compression.swift:345]: Fixed

**Comment by valeriyvan** [comment]
Compression was removed from master

**Pull request closed without merging**